src/screens/splash.rs

```diff
@@ -35,48 +39,82 @@ pub(super) fn plugin(app: &mut App) {
         )
             .run_if(in_state(Screen::Splash)),
     );
+
+    // Exit the splash screen early if the player hits escape.
+    app.add_systems(
+        Update,
+        exit_splash_screen
+            .run_if(input_just_pressed(KeyCode::Escape).and_then(in_state(Screen::Splash))),
+    );
 }
 
 const SPLASH_BACKGROUND_COLOR: Color = Color::srgb(0.157, 0.157, 0.157);
 const SPLASH_DURATION_SECS: f32 = 1.8;
 const SPLASH_FADE_DURATION_SECS: f32 = 0.6;
 
-fn spawn_splash(mut commands: Commands, asset_server: Res<AssetServer>) {
-    commands
-        .ui_root()
-        .insert((
-            Name::new("Splash screen"),
+#[derive(Component, Reflect)]
+#[reflect(Component)]
+struct SplashScreenContainer;
+
+#[derive(Resource, Reflect)]
+#[reflect(Resource)]
+struct SplashScreenImageList(VecDeque<Handle<Image>>);
+
+impl FromWorld for SplashScreenImageList {
+    fn from_world(world: &mut World) -> Self {
+        let asset_server = world.resource::<AssetServer>();
+
+        // To show your own splash images, replace or add images here.
+        Self(VecDeque::from_iter(
+            [asset_server.load("images/splash.png")],
```

Comment:
This should keep the original `load_with_settings` and comments from before the PR.